In [2]:
import pandas as pd

In [3]:
cust = pd.read_csv('../data/source/customer_master_degraded.csv')
cust.shape

(102440, 23)

In [4]:
cust.head()

,customer_id,first_name,last_name,street,city,region,postal_code,country,phone,email,...,distribution_channel,division,currency,payment_terms,credit_limit,language,created_date,gender,age,payment_method
0,C319333,Allison,Benson,6780 Victor Street,East Sylviaport,Cristianview,68969,FR,+1 769-768-4617,allison.benson@outlook.com,...,20,0,EUR,NT00,0.0,EN,03/12/2020,Female,29.0,Cash
1,C882327,Çetinsu,Akçay,1801 Karadeniz Island Suite 495,Antalya,Black Sea,46114,TR,+90 521 449 6506,çetinsu.akçay@hotmail.com,...,20,10,TRY,NT60,5000.0,TR,05/12/2024,Female,46.0,Cash
2,C112124,Galina,Gieß,Volkhard-Caspar-Weg 6260,Eisenberg,Thüringen,76069,DE,+49 164 3830571,galina.gieß@hotmail.com,...,10,0,EUR,NT00,0.0,DE,2023-09-26,Male,62.0,Credit Card
3,C341690,Cuheyna,Manço,693 Kasim Locks Suite 929,Konya,Black Sea,61222,TR,+90 505 737 5296,cuheyna.manço@outlook.com,...,10,10,TRY,NT30,0.0,TR,2022-10-29,Female,38.0,Cash
4,C309733,Thomas,Yates,598 Crystal Shores,Morahaven,New Amandaton,3837,FR,+1 634-291-8165,thomas.yates@icloud.com,...,10,0,EUR,NT15,0.0,EN,2023-07-07,female,50.0,Cash


In [5]:
sales = pd.read_csv('../data/source/sales_data_degraded.csv',low_memory=False)
sales.shape

(100650, 7)

In [6]:
sales.head()

,invoice_no,customer_id,category,quantity,price,invoice_date,shopping_mall
0,I180920,C258332,Souvenir,1,11.73,23-10-2022,Mall of Istanbul
1,I228446,C892382,Clothing,1,300.08,22-08-2021,Istinye Park
2,I128686,C214740,Clothing,2,600.16,08-07-2021,Istinye Park
3,I714478,C278988,SOUVENIR,1,11.73,18-04-2022,Emaar Square Mall
4,I159136,C135574,Food & Beverage,3,15.69,22-10-2022,Metrocity


In [7]:
(cust.isnull().mean()*100).round(2).sort_values(ascending=False)

phone                   6.64
email                   5.72
region                  5.64
tax_id                  4.49
credit_limit            3.03
last_name               1.00
first_name              0.99
country                 0.50
age                     0.12
customer_id             0.00
currency                0.00
gender                  0.00
created_date            0.00
language                0.00
payment_terms           0.00
account_group           0.00
division                0.00
distribution_channel    0.00
sales_org               0.00
postal_code             0.00
city                    0.00
street                  0.00
payment_method          0.00
dtype: float64

In [15]:
cust['country'].value_counts()

country
TR                68641
DE                 8852
US                 4401
GB                 2641
turkey             2149
tr                 2146
TUR                2112
TÜRKİYE            2111
Türkiye            2102
Turkey             2067
FR                 1721
germany             313
Deutschland         300
DEU                 300
Germany             296
GERMANY             294
America             169
U.S.A.              146
us                  144
United States       141
USA                 135
England              91
United Kingdom       90
Great Britain        89
FRANCE               86
gb                   83
UK                   83
FRA                  80
France               75
france               72
Name: count, dtype: int64

In [9]:
cust['gender'].value_counts()

gender
Female    56376
Male      37835
female      731
FEMALE      712
F           701
2           700
Woman       697
F           682
f           651
m           498
1           491
Man         482
M           476
male        475
M           475
MALE        458
Name: count, dtype: int64

In [10]:
cust['phone'].value_counts().head(20)

phone
Unknown             277
-                   259
TBD                 250
?                   246
                    230
+90 598 784 6115      3
+90 567 930 9726      2
+90 571 228 6477      2
+90-509-151-3400      2
(578) 861-3028        2
+49 172 9993605       2
+90 543 369 6331      2
+90 526 544 3570      2
+90 580 375 4164      2
+90 574 717 2272      2
+90 508 416 6606      2
+90 507 815 1193      2
+90 512 604 9015      2
+90 519 243 8697      2
+1 384-585-6696       2
Name: count, dtype: int64

In [11]:
cust['age'].describe()

count    102321.000000
mean         46.357151
std          53.386138
min          -5.000000
25%          30.000000
50%          43.000000
75%          57.000000
max         999.000000
Name: age, dtype: float64

In [12]:
cust['credit_limit'].describe()

count    99341.000000
mean      5363.666563
std      10308.294249
min     -50000.000000
25%          0.000000
50%       1000.000000
75%       5000.000000
max      50000.000000
Name: credit_limit, dtype: float64

In [17]:
int(cust.duplicated().sum())

1791

In [22]:
(cust['credit_limit'] < 0).sum()

np.int64(999)

Data Integrity Assessment

Missing Data
Worst categories: phone(6.64%), email(5.72%), region(5.64%)

Standardization Snapshot
7 different variants for Turkey alone
30 unique country designations
Multiple representations of M/F gender alone

Value Sanity
Age sees multiple indicators of incorrect data entry
Credit alone has 999 entries with a negative credit limit

Duplicate Check
1791 exact duplicates

Concerns
Country remediation is a long-term concern going forward